docs/cvl/functions.md

```diff
@@ -43,3 +43,77 @@ Using CVL functions
 -------------------
   CVL Function may be called from within a rule, or from within another CVL function.
 
+Math and rounding summaries
+---------------------------
+
+In real protocols, arithmetic often requires precise rounding envelopes. CVL functions are a clean way to centralize these rules as small, reusable summaries.
+
+- Round-aware `mulDiv` abstraction with explicit direction:
+
+  ```cvl
+  // Rounds up or down depending on a Math.Rounding enum
+  function mulDivCVL(uint256 x, uint256 y, uint256 denominator, Math.Rounding rounding) returns uint256 {
+      if (rounding == Math.Rounding.Floor) {
+          return mulDivDownAbstractPlus(x, y, denominator);
+      } else {
+          return mulDivUpAbstractPlus(x, y, denominator);
+      }
+  }
+  ```
+
+- Tight, solver-friendly models for up/down `mulDiv` used across specs:
+
+  ```cvl
+  function mulDivDownAbstractPlus(uint256 x, uint256 y, uint256 z) returns uint256 {
+      uint256 res;
+      require z != 0;
+      uint256 xy = require_uint256(x * y);
+      uint256 fz = require_uint256(res * z);
+      require xy >= fz;
+      require fz + z > to_mathint(xy);
+      return res;
+  }
+
+  function mulDivUpAbstractPlus(uint256 x, uint256 y, uint256 z) returns uint256 {
+      uint256 res;
+      require z != 0;
+      uint256 xy = require_uint256(x * y);
+      uint256 fz = require_uint256(res * z);
+      require xy >= fz;
+      require fz + z > to_mathint(xy);
+      if (xy == fz) { return res; }
+      return require_uint256(res + 1);
```

Comment:
I would appreciate someone using thses summaries daily to say whether they're really good